python/pyarrow/table.pxi

```diff
@@ -5042,9 +5043,14 @@ def concat_tables(tables, c_bool promote=False, MemoryPool memory_pool=None):
     tables : iterable of pyarrow.Table objects
         Pyarrow tables to concatenate into a single Table.
     promote : bool, default False
-        If True, concatenate tables with null-filling and null type promotion.
+        If True, concatenate tables with null-filling and type promotion.
+        See field_merge_options for the type promotion behavior.
```

Comment:
From a user experience perspective, I find it a bit strange that I still have to specify `promote=True` when I already specified field_merge_options="permissive" (because the latter only makes sense with promote=True). Essentially the `promote=False` can be seen as a "no promotion" field merge option? 

I know this is an existing keyword, but maybe going forward we should try to merge `promote` and `field_merge_options` in a single keyword? 

(on the naming side, I think I actually like "promote_options" better than "field_merge_options", but that would then deviate from the C++ naming ..)

Reply:
> Essentially the promote=False can be seen as a "no promotion" field merge option?

This is correct, it will only do null-filling for missing columns.

The problem is that the `promote` is only relevant when merging tables, when just upcasting fields, then you don't need this, therefore `merge_options` might not be the best place to put it.

Reply:
> The problem is that the `promote` is only relevant when merging tables, when just upcasting fields, then you don't need this, therefore `merge_options` might not be the best place to put it.

Do you mean with "merging" tables for when the columns don't match? So if a column is not present in one of the tables, it gets filled with all nulls. This specific aspect is not something that you can control with `field_merge_options`? (you can only control whether to upcast null type to the target type, if the nulls where already present)